moxin-frontend/src/data/store.rs

```diff
@@ -49,6 +49,7 @@ pub struct DownloadInfo {
 pub struct ModelWithPendingDownloads {
     pub model: Model,
     pub pending_downloads: Vec<PendingDownload>,
+    pub current_file_id: Option<FileID>,
```

Comment:
I might be missing something but can you use the already existing `current_chat` in `Store` instead of adding this to `ModelWithPendingDownloads`? Since in `draw_walk` of `ModelFilesList` you have access to the Store. It might  look like more work on the other side but semantically it doesn't seem like `ModelWithPendingDownloads` would need a `current_file_id`.

Reply:
I'm almost sure there is no way to reach the `store` from `ModelFilesList`. This is because we're passing more granular data at this widget (represented with the struct `ModelWithPendingDownloads`). So, I think @noxware was following the existing approach there.

Reply:
Relevant code: https://github.com/project-robius/moxin/blob/main/moxin-frontend/src/landing/model_list.rs#L101-L103

(from this point deeper in the widgets tree, the scope does not hold the entire `store`)

Reply:
Yes, what @jmbejar mentioned is the reason why I choose to do it this way. The `draw_walk` and `handle_event` receive scopes with different things. The draw walk receives what @jmbejar mentioned.

I think the store is accesible in the handle event (which surprised me and confused me a lot until I followed the code up to discover the origin of both scopes), but adding a `#rust` field and setting it on events as a workaround was worst in my opinion.

Extra note:
I noticed that makepad, apart of having a `Scope.with_data` has a `Scope.with_props`. The difference is that one is mutable and the later is immutable. If they are hold in different slots, then, would this problem be resolved by using `props` for `ModelWithPendingDownloads` and reserve `data` to always pass the global context (store) down?